### PR TITLE
Feature/rate weet

### PR DESCRIPTION
**Pull request by PedrOrtizM**
## Summary
- POST: /weets/:id/ratings end point
- Validations
- Controller and services
- Unit test

## Trello Card
https://trello.com/c/pBJSMEhd/37-calificar-weet

### CODE REVIEW

File: app/helpers/calculate-user-score.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module.exports.calculateUserScore = (newScore, currentScore) => {
+  if (currentScore != 0)
+    return newScore === currentScore ? newScore * -1 : newScore * 2;
```
Comment (dbenavidesv) [inline app/helpers/calculate-user-score.js:3]: Why this? can you explain?

File: app/services/rating.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+const { Rating, sequelize } = require('../models');
+const logger = require('../logger');
+const { databaseError } = require('../errors');
+const { calculateUserScore } = require('../helpers/calculate-user-score');
+
+exports.createRating = async (rating, user) => {
+  const transaction = await sequelize.transaction();
```
Comment (dbenavidesv) [inline app/services/rating.js:7]: This should be inside the catch

File: app/services/rating.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    return ratingInfo;
+  } catch (error) {
+    logger.error(error);
+    if (transaction.rollback) await transaction.rollback();
+    throw databaseError(error);
+  }
+};
+
+exports.updateRating = async (score, rating, user) => {
+  const transaction = await sequelize.transaction();
```
Comment (dbenavidesv) [inline app/services/rating.js:24]: same here

File: app/helpers/calculate-user-score.js (hunk 1 above)
Comment (PedrOrtizM) [reply app/helpers/calculate-user-score.js:3]: It is for these requirements:
image

- If you want to set the same value then you have to remove the current adding the opposite value. `newScore * -1`.
- else you have to remove the old value and add the new one. `newScore * 2`.

Comment (dbenavidesv) [reply app/helpers/calculate-user-score.js:3]: but why are you multiplying?

Comment (PedrOrtizM) [reply app/helpers/calculate-user-score.js:3]: To set the sign and add or substract in the service. If the `currentScore =1` and the `newScore=1`( are the same value ) then i need substract 1 to user score `newScore * -1`  ` 1 * -1 =  -1`.  then `newUserScore =-1`. in the  service substract 1: 
`await user.update({ score: (newUserScore + currentUserScore) }, { transaction });`
the same when newScore is negative.
I think I could improve the variables names to be clear.

File: app/services/rating.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    throw databaseError(error);
+  }
+};
+
+exports.updateRating = async (score, rating, user) => {
+  try {
+    const transaction = await sequelize.transaction();
+    const { score: currentScore } = rating;
+    const { score: currentUserScore } = user;
+    const newRatingScore = currentScore === score ? 0 : score;
```
Comment (dbenavidesv) [inline app/services/rating.js:28]: If the new score is the same a the current one, just do nothing and return. I think you are over complicating things here 😅

**Review by dbenavidesv — APPROVED** [review]

**Pull request merged**